### PR TITLE
Add more debug log methods to sm_logging python module

### PR DESCRIPTION
**Pull request by uschwes**
* Add Fine, Trace, Finer, Verbose, Finest and All log methods
* Removed code duplication

### CODE REVIEW

**Comment by uschwes** [comment]
retest this please

**Comment by uschwes** [comment]
@HannesSommer I think something is broken with osx in master already...

**Comment by HannesSommer** [comment]
On Ubuntu it is the missing latex it got executed on mc-jenkins and that it lacking latex. Same for the mac. At some point we should try to get rid of this stupid latex dependency :).

**Comment by uschwes** [comment]
http://129.132.38.183:8080/job/SchweizerMesser/133/label=osx/testReport/junit/%28root%29/QuaternionAlgebraTestSuite/testAxisAngle2QuatAndBack/

**Comment by HannesSommer** [comment]
Ah I you were referring to the tests. Yes, strange I don't know what makes the difference there. But I guess it is safe to raise it a bit. I just did so with PR #134.

**Comment by uschwes** [comment]
retest this please

**Comment by uschwes** [comment]
retest this please

**Comment by uschwes** [comment]
Works as expected after rigorous testing on own machine.
@HannesSommer good to merge even if jenkins did not confirm yet due to other problems?

File: sm_python/python/sm/__init__.py [hunk 1]
```diff
@@ -8,101 +8,85 @@
...
+    if getLoggingLevel() > level:
         return
-    callerframerecord = inspect.stack()[1]
+    callerframerecord = inspect.stack()[stackdepth]
     frame = callerframerecord[0]
     info = inspect.getframeinfo(frame)
-    rawLog(LoggingLevel.Debug, info.filename, info.lineno, info.function, message)
+    rawLogNamed(name,level, info.filename, info.lineno, info.function, message)
+
+def logAll(message):
```
Comment (HannesSommer) [inline sm_python/python/sm/__init__.py]: Remove this, it's confusing :).

File: sm_python/python/sm/__init__.py [hunk 2]
```diff
@@ -8,101 +8,85 @@
...
-        return
-    callerframerecord = inspect.stack()[1]
-    frame = callerframerecord[0]
-    info = inspect.getframeinfo(frame)
-    rawLog(level, info.filename, info.lineno, info.function, message)
-
+    log(LoggingLevel.Error, message, stackdepth=2)
+
+
+def logAllNamed(name, message):
```
Comment (HannesSommer) [inline sm_python/python/sm/__init__.py]: Remove this too.

**Pull request merged**